clients/js/genjs.go

```diff
@@ -243,10 +243,19 @@ var methodTmplStr = `
     const timeout = options.timeout || this.timeout;
     const span = options.span;
 
-    const headers = {};{{range $param := .HeaderParams}}
-    headers["{{$param.WagName}}"] = params.{{$param.JSName}};{{end}}
-
-    const query = {};{{range $param := .QueryParams}}{{ if $param.Required }}
+    const headers = {};
+    {{- range $param := .PathParams}}
+    if (params.{{$param.JSName}} == "") {
```

Comment:
Could do this just as `if (!params.{{$param.JSName}}) {` which would catch the undefined case as well. 

If we don't do that we should use `===` I think.